tests/unit/test_gcstorage.py

```diff
@@ -33,89 +33,89 @@ def test_split_path(self):
-        mock.return_value = [m]
-        assert not self.s.file_exists('/mybucket/file')
-        mock.assert_called_with('/mybucket/file', delimiter='/')
-
-    @patch('gaepypi.storage.gcs.listbucket')
-    def test_path_is_dir(self, mock):
-        m = Mock()
-        type(m).is_dir = PropertyMock(return_value=True)
-        mock.return_value = [m]
-        assert self.s.path_exists('/mybucket/file')
-        mock.assert_called_with('/mybucket/file', delimiter='/')
-
-    @patch('gaepypi.storage.gcs.listbucket')
-    def test_path_exists(self, mock):
-        m = Mock()
-        type(m).is_dir = PropertyMock(return_value=False)
-        mock.return_value = [m]
-        assert self.s.path_exists('/mybucket/file.txt')
-        mock.assert_called_with('/mybucket/file.txt', delimiter='/')
-
-    @patch('gaepypi.storage.gcs.listbucket')
-    def test_ls_all(self, mock):
-        files = []
-        filenames = list(map(lambda f: '/mybucket/path/{0}'.format(f), ['dummy', 'a.txt', 'b.txt', 'wheel']))
-        for isdir, filename in zip([True, False, False, True], filenames):
-            m = Mock()
-            type(m).is_dir = PropertyMock(return_value=isdir)
-            type(m).filename = PropertyMock(return_value=filename)
-            files.append(m)
-        mock.return_value = files
-        retrieved = self.s.ls('/mybucket/path/')
-        mock.assert_called_with('/mybucket/path/', delimiter='/')
-        assert len(retrieved) == 4
-        assert filenames == retrieved
-
-    @patch('gaepypi.storage.gcs.listbucket')
-    def test_ls_dir_only(self, mock):
-        files = []
-        filenames = list(map(lambda f: '/mybucket/path/{0}'.format(f), ['dummy', 'a.txt', 'b.txt', 'wheel']))
-        for isdir, filename in zip([True, False, False, True], filenames):
-            m = Mock()
-            type(m).is_dir = PropertyMock(return_value=isdir)
-            type(m).filename = PropertyMock(return_value=filename)
-            files.append(m)
-        mock.return_value = files
-        retrieved = self.s.ls('/mybucket/path', dir_only=True)
-        mock.assert_called_with('/mybucket/path/', delimiter='/')
-        assert len(retrieved) == 2
-        assert filenames[0] == retrieved[0]
-        assert filenames[3] == retrieved[1]
-
-    @patch('gaepypi.storage.gcs.open')
-    def test_open(self, mock):
-        self.s.read('/mybucket/path/a.txt')
-        mock.assert_called_with('/mybucket/path/a.txt')
-
-    @patch('gaepypi.storage.gcs.open')
-    @patch('gaepypi.storage.gcs.RetryParams')
-    def test_write(self, retry, open):
-        retry.return_value = 'rp'
-        m = Mock()
-        m.write = Mock()
-        open.return_value = m
-        self.s.write('/mybucket/path/a.txt', '1111111')
-        retry.assert_called_with(backoff_factor=1.1)
-        open.assert_called_with('/mybucket/path/a.txt', 'w', options={'x-goog-acl': 'project-private'}, retry_params='rp')
-        m.write.assert_called_with('1111111')
+    #
```

Comment:
why like this? we love tests :)

Reply:
this is old lib that is no longer relevant